### DISCUSSION TITLE
What is a callback protocol?

### DISCUSSION THREAD

**Discussion by superbobry** [python/typing#1432] [Q&A]
Is it a protocol with exactly a `__call__` method or a protocol with *at least* a `__call__` method?

The relevant [section](https://peps.python.org/pep-0544/#callback-protocols) in PEP-544 does not clarify that, unfortunately.

**Comment by Akuli** [comment]
A callback protocol only has `__call__`. The intent is to specify that the object should be callable in a specific way, with no other requirements. You can include other things too, of course, but it doesn't make sense in the context of the PEP, because it is describing how to work around the limitations of `Callable`.

Maybe the text in the PEP should be clarified :)

**Reply by AlexWaygood** [reply]
The general feeling is that PEPs should be seen as historical documents rather than pieces of living documentation, so a PR to improve the wording of PEP 544 probably wouldn't be accepted at this stage. I recently added a new section to the CPython typing docs on `Callable` and callback protocols, however, and PRs to improve that section would definitely be accepted: https://docs.python.org/3/library/typing.html#annotating-callable-objects

**Reply by superbobry** [reply]
Alex, this is off topic, but I think we should have a separate "living" document which specifies how all type system features work together. Standard library documentation is aimed at typing users, but that document would be aimed at tooling authors.

**Reply by AlexWaygood** [reply]
Fully agreed that that would be nice in principle. A new section in https://typing.readthedocs.io/en/latest/index.html might work well for that purpose.

**Reply by hauntsaninja** [reply]
You mean like https://typing.readthedocs.io/en/latest/source/protocols.html#callback-protocols ;-)

**Reply by AlexWaygood** [reply]
> You mean like https://typing.readthedocs.io/en/latest/source/protocols.html#callback-protocols ;-)

I'm a fan! But that still feels like user-oriented docs, whereas @superbobry here seems to be asking for a separate set of docs aimed at tooling authors
